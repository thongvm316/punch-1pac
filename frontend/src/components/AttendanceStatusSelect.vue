<template>
  <select class="form-select" @change="updateValue($event)">
    <slot name="placeholder"></slot>
    <option :value="status" v-for="status in meta.attendance_statuses">{{ $t(`meta.attendance_statuses.${status}`) }}</option>
  </select>
</template>

<script>
import { mapState } from 'vuex'

export default {
  name: 'attendance-status-select',

  props: ['value'],

  computed: {
    ...mapState('initialStates', ['meta'])
  },

  methods: {
    updateValue(e) {
      this.$emit('input', e.target.value)
    }
  }
}
</script>
