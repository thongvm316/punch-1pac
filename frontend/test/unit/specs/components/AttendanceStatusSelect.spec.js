import { shallowMount } from '@vue/test-utils'
import wrapperOps from '../../supports/wrapper'
import setComputed from '../../supports/set-computed'
import initialStatesData from '../../supports/fixtures/initial-states.api'
import AttendanceStatusSelect from '@/components/AttendanceStatusSelect'

const updateValue = jest.spyOn(AttendanceStatusSelect.methods, 'updateValue')
const scopedSlots= {
  placeholder: '<option>Attendance status</option>'
}

Object.assign(wrapperOps, {
  scopedSlots
})

describe('AttendanceStatusSelect.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(AttendanceStatusSelect, wrapperOps)
    setComputed(wrapper, {
      meta: initialStatesData.meta
    })
  })

  afterEach(() => { wrapper.vm.$destroy() })

  describe('when AttendanceStatusSelect was mounted', () => {
    it('should render correctly', () => {
      expect(wrapper.exists()).toBe(true)
      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when onChange method', () => {
    it('should call onChange method', () => {
      wrapper.find('select.form-select').trigger('change')

      expect(updateValue).toHaveBeenCalled()
      expect(wrapper.emitted().input).toBeTruthy()
      expect(wrapper.emitted().input).toHaveLength(1)
    })

    it('should emtting correctly data', () => {
      wrapper.vm.$emit('input', 'attend_ok')

      expect(wrapper.emitted().input[0]).toEqual(['attend_ok'])
    })
  })
})
