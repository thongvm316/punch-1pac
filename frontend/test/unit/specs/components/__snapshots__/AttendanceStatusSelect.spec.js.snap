// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttendanceStatusSelect.vue when AttendanceStatusSelect was mounted should render correctly 1`] = `
<select class="form-select">
  <option>Attendance status</option>
  <option value="attend_ok">
    Attend OK
  </option>
  <option value="attend_late">
    Attend Late
  </option>
  <option value="leave_ok">
    Leave OK
  </option>
  <option value="leave_early">
    Leave Early
  </option>
  <option value="annual_leave">
    Day Off
  </option>
</select>
`;
